processing/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java

```diff
@@ -100,13 +100,15 @@ public static void throwIfInvalidProtocols(HttpInputSourceConfig config, List<UR
 
   public static void throwIfForbiddenHeaders(HttpInputSourceConfig config, Map<String, String> requestHeaders)
   {
-    if (config.getAllowedHeaders().size() > 0) {
-      for (Map.Entry<String, String> entry : requestHeaders.entrySet()) {
-        if (!config.getAllowedHeaders().contains(StringUtils.toLowerCase(entry.getKey()))) {
-          throw InvalidInput.exception("Got forbidden header %s, allowed headers are only %s ",
-                                       entry.getKey(), config.getAllowedHeaders()
-          );
-        }
+    String message = null;
+    if (config.getAllowedHeaders().size() == 0 && requestHeaders.size() > 0) {
+      message = "You can set the property druid.ingestion.http.allowedHeaders in middle managers or peons to whitelist request headers";
+    }
```

Comment:
this block can be removed while keeping the functionality just the same as before.